### PR TITLE
Trit 61: Added changes for dynamically rendering the tax-year

### PR DESCRIPTION
**Pull request by suryend4u**
Trit 61: Added changes for dynamically rendering the tax-year

### CODE REVIEW

File: app/controllers/HomePageController.scala [hunk 1]
```diff
@@ -46,7 +47,7 @@ class HomePageController @Inject()(
     implicit request =>
 
       sessionRepository.set(UserAnswers(request.internalId)).map{ _ =>
-        Redirect(routes.HomePageController.onPageLoad())
+        Redirect(routes.DaysSpentInUKController.onPageLoad(NormalMode))
```
Comment (abhijeetardale) [inline app/controllers/HomePageController.scala]: please use Navigators for redirect

File: app/models/RelevantTaxYear.scala [hunk 2]
```diff
@@ -15,19 +15,34 @@
...
 
-object RelevantTaxYear extends Enumerable.Implicits {
+object RelevantTaxYear extends Enumerable.Implicits with CurrentTaxYear{
 
-  case object Py extends WithName("pY") with RelevantTaxYear
-  case object Py1 extends WithName("pY-1") with RelevantTaxYear
-  case object Py2 extends WithName("pY-2") with RelevantTaxYear
-  case object Py3 extends WithName("pY-3") with RelevantTaxYear
+  override def now = () => DateTime.now
+  val cTaxYear = taxYearFor(today)
```
Comment (abhijeetardale) [inline app/models/RelevantTaxYear.scala]: Please check with Ankit before as he has refactored this code for common DateUtils

File: test/views/NonTaxResidentResultSpec.scala [hunk 3]
```diff
@@ -34,6 +35,8 @@ import uk.gov.hmrc.viewmodels.NunjucksSupport
 import views.behaviours.ViewBehaviours
 
 class NonTaxResidentResultSpec extends SpecBase with MockitoSugar with NunjucksSupport with JsonMatchers with ViewBehaviours{
+  val formProvider = new DaysSpentInUKFormProvider()
```
Comment (abhijeetardale) [inline test/views/NonTaxResidentResultSpec.scala]: please remove unused references

File: app/models/RelevantTaxYear.scala (hunk 2 above)
Comment (suryend4u) [reply app/models/RelevantTaxYear.scala]: Will resolve this class while merging.

File: test/views/NonTaxResidentResultSpec.scala (hunk 3 above)
Comment (suryend4u) [reply test/views/NonTaxResidentResultSpec.scala]: Done

File: conf/app.routes [hunk 4]
```diff
@@ -23,7 +23,7 @@ GET        /non-tax-resident-result                     controllers.NonTaxReside
 
 GET        /tax-resident-result                         controllers.TaxResidentResultController.onPageLoad()
 
-GET        /days-in-uk                                  controllers.DaysSpentInUKController.onPageLoad(mode: Mode = NormalMode)
-POST       /days-in-uk                                  controllers.DaysSpentInUKController.onSubmit(mode: Mode = NormalMode)
-GET        /change-days-in-uk                           controllers.DaysSpentInUKController.onPageLoad(mode: Mode = CheckMode)
-POST       /change-days-in-uk                           controllers.DaysSpentInUKController.onSubmit(mode: Mode = CheckMode)
+GET        /days-spent-in-UK                            controllers.DaysSpentInUKController.onPageLoad(mode: Mode = NormalMode)
```
Comment (abhijeetardale) [inline conf/app.routes]: url should be small letters

File: app/controllers/HomePageController.scala (hunk 1 above)
Comment (suryend4u) [reply app/controllers/HomePageController.scala]: Ankit has changed it already. So not changing it again. Line 50 should be taken from his code.

**Pull request merged**